src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```
@@ -0,0 +1,313 @@
+                }
+
+                try
+                {
+                    var clonedParameters = nodeParameters.Clone();
+
+                    Parallel.ForEach(peersToDiscover, new ParallelOptions()
+                    {
+                        MaxDegreeOfParallelism = 5,
+                        CancellationToken = nodeParameters.ConnectCancellation,
+                    },
+                    peer =>
+                    {
+                        using (var connectTokenSource = CancellationTokenSource.CreateLinkedTokenSource(nodeParameters.ConnectCancellation))
+                        {
+                            connectTokenSource.CancelAfter(TimeSpan.FromSeconds(5));
+
+                            Node node = null;
+
+                            try
+                            {
+                                clonedParameters.ConnectCancellation = connectTokenSource.Token;
+
+                                var addressManagerBehaviour = clonedParameters.TemplateBehaviors.Find<PeerAddressManagerBehaviour>();
+                                clonedParameters.TemplateBehaviors.Clear();
+                                clonedParameters.TemplateBehaviors.Add(addressManagerBehaviour);
+
+                                node = Node.Connect(network, peer.Endpoint, clonedParameters);
+                                node.VersionHandshake(connectTokenSource.Token);
+                                node.MessageReceived += (s, a) =>
+                                {
+                                    if (a.Message.Payload is AddrPayload addressPayload)
+                                        Interlocked.Add(ref peersFound, addressPayload.Addresses.Length);
+                                };
+
+                                node.SendMessageAsync(new GetAddrPayload());
+
+                                connectTokenSource.Token.WaitHandle.WaitOne(2000);
+                            }
+                            catch
```

Redundant empty catch